internal/indexer/watched_accounts_cache.go

```diff
@@ -0,0 +1,94 @@
+package indexer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/bits-and-blooms/bloom/v3"
+	"github.com/stellar/go/support/log"
+)
+
+type ReloadFn func(ctx context.Context) ([]string, error)
+
+type WatchedAccountsCache struct {
+	bloomFilter    atomic.Value // holds *bloom.BloomFilter
+	reloadInterval time.Duration
+	reloadFn       ReloadFn
+	quit           chan struct{}
+}
+
+func NewWatchedAccountsCache(ctx context.Context, reloadInterval time.Duration, reloadFn ReloadFn) *WatchedAccountsCache {
+	wc := &WatchedAccountsCache{
+		reloadInterval: reloadInterval,
+		reloadFn:       reloadFn,
+		quit:           make(chan struct{}),
+	}
+	go wc.periodicRebuild(ctx)
+	return wc
+}
+
+// periodicRebuild is a goroutine that periodically reloads the watched accounts and updates the bloom filter.
+// It also listens for signals to stop the goroutine.
+// It is started by NewWatchedAccountsCache and can be stopped by Close().
+func (wc *WatchedAccountsCache) periodicRebuild(ctx context.Context) {
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, syscall.SIGTERM, syscall.SIGINT, syscall.SIGQUIT)
+	defer signal.Stop(signalChan)
+
+	ticker := time.NewTicker(wc.reloadInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Ctx(ctx).Infof("watched accounts cache stopped due to context cancellation: %v", ctx.Err())
+			return
+
+		case sig := <-signalChan:
+			log.Ctx(ctx).Warnf("watched accounts cache stopped due to signal %s", sig)
+			return
+
+		case <-wc.quit:
+			log.Ctx(ctx).Infof("watched accounts cache stopped due to quit signal")
+			return
+
+		case <-ticker.C:
+			wc.DoRebuild(ctx)
+		}
+	}
+}
```

Comment:
The bloom filter will periodically rebuild using the provided method. We'll use `AccountsModel.GetAll` to rebuild it.